### ISSUE TITLE
Negative sender balance throws error in `_executeCall`

### ISSUE THREAD

**Issue by acolytec3** [ethereumjs/ethereumjs-monorepo#1839]
On `develop`, if `_executeCall` is executed during `executeMessage` and the sender's balance becomes negative, instead of correctly reverting the call, an error as shown below is thrown because the `bigIntToBuffer` doesn't correctly account for negative values.

```sh
Error: Cannot convert string to buffer. toBuffer only supports 0x-prefixed hex strings and this string was given: 0x-1
  at toBuffer (/home/jim/development/ethereumjs-monorepo/packages/util/src/bytes.ts:168:13)
  at bigIntToBuffer (/home/jim/development/ethereumjs-monorepo/packages/util/src/bytes.ts:224:18)
  at bigIntToUnpaddedBuffer (/home/jim/development/ethereumjs-monorepo/packages/util/src/bytes.ts:370:22)
  at Account.raw (/home/jim/development/ethereumjs-monorepo/packages/util/src/account.ts:86:29)
  at Account.serialize (/home/jim/development/ethereumjs-monorepo/packages/util/src/account.ts:96:28)
  at Cache._update (/home/jim/development/ethereumjs-monorepo/packages/vm/src/state/cache.ts:189:23)
```

Theoretically, this can only happen in `runTx` if the `skipBalance` parameter is passed to the function call but since we don't currently handle negative `bigint` to `buffer` conversions in our code, it throws instead of proceeding with function execution.

One possible solution would be to add add an arbitrarily high balance to the `from` account if `skipBalance` is set and then subtract out the added balance before returning the execution result (or set to 0 if it would be negative).  We also need to handle the same situation in `runCall` as a similar error will occur if the balance ever goes negative.

**Comment by acolytec3** [comment]
It's not obvious to me why this doesn't throw an error on `master` but it's something related to how `BN.js` handles negative numbers.

**Comment by holgerd77** [comment]
Oh, this might need some deeper analysis on negative number behavior in the VM (eventually, other libs as well) along with this BigInt switch? 🤔

**Comment by holgerd77** [comment]
Closed by #1849
